controller/src/main/java/org/jboss/as/controller/extension/ExtensionResource.java

```diff
@@ -248,4 +248,15 @@ public PlaceholderResourceEntry clone() {
     public Set<String> getOrderedChildTypes() {
         return Collections.emptySet();
     }
+
+    @Override
+    public int hashCode() {
+        return this.moduleName.hashCode();
```

Comment:
why are you not taking `extensionRegistry` into account?

Reply:
Because the path element of an extension is sufficient to identify uniqueness.